src/System.Runtime/tests/System/Version.cs

```diff
@@ -3,211 +3,263 @@
     {
-        var version = new Version(major, minor, build);
-        VerifyVersion(version, major, minor, build, -1);
+        VerifyVersion(new Version(major, minor, build, revision), major, minor, build, revision);
     }
 
     [Fact]
-    public void TestMajorMinorBuildRevisionConstructor()
+    public static void TestCtor_Int_Int_Int_Int_Invalid()
     {
-        VerifyConstructor(0, 0, 0, 0);
-        VerifyConstructor(2, 3, 4, 7);
-        VerifyConstructor(2, 3, 4, 65535);
-        VerifyConstructor(2, 3, 4, 65536);
-        VerifyConstructor(2, 3, 4, 32767);
-        VerifyConstructor(2, 3, 4, 32768);
-        VerifyConstructor(2, 3, 4, 2147483647);
-        VerifyConstructor(2, 3, 4, 2147450879);
-        VerifyConstructor(2, 3, 4, 2147418112);
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Version(-1, 0, 0, 0));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Version(0, -1, 0, 0));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Version(0, 0, -1, 0));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Version(0, 0, 0, -1));
+        Assert.Throws<ArgumentOutOfRangeException>("major", () => new Version(-1, 0, 0, 0)); // Major < 0
+        Assert.Throws<ArgumentOutOfRangeException>("minor", () => new Version(0, -1, 0, 0)); // Minor < 0
+        Assert.Throws<ArgumentOutOfRangeException>("build", () => new Version(0, 0, -1, 0)); // Build < 0
+        Assert.Throws<ArgumentOutOfRangeException>("revision", () => new Version(0, 0, 0, -1)); // Revision < 0
     }
 
-    private static void VerifyConstructor(int major, int minor, int build, int revision)
+    public static IEnumerable<object[]> CompareToTestData()
     {
-        var version = new Version(major, minor, build, revision);
-        VerifyVersion(version, major, minor, build, revision);
+        yield return new object[] { new Version(1, 2), null, 1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 2), 0 };
+        yield return new object[] { new Version(1, 2), new Version(2, 0), -1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 2, 1), -1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 2, 0, 1), -1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 0), 1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 0, 1), 1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 0, 0, 1), 1 };
+
+        yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 4), 0 };
+        yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 5), -1 };
+        yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 3), 1 };
     }
 
-    private static void VerifyVersion(Version version, int major, int minor, int build, int revision)
+    [Theory, MemberData(nameof(CompareToTestData))]
```

Comment:
Almost everywhere else you've used
```C#
[Theory]
[MemberData(nameof(SomeMethod))]
```

but here you used the one-line version.  Why is this one different?